GVFS/GVFS.Common/Git/GitProcess.cs

```diff
@@ -405,11 +404,21 @@ public Process GetGitProcess(string command, string workingDirectory, string dot
 
             // Removing trace variables that might change git output and break parsing
             // List of environment variables: https://git-scm.com/book/gr/v2/Git-Internals-Environment-Variables
-            foreach (string key in processInfo.EnvironmentVariables.Keys.Cast<string>()
-                .Where(x => x.StartsWith("GIT_TRACE", StringComparison.OrdinalIgnoreCase))
-                .ToList())
+            foreach (string key in processInfo.EnvironmentVariables.Keys.Cast<string>())
             {
-                processInfo.EnvironmentVariables.Remove(key);
+                // If GIT_TRACE is set to a fully-rooted path, then Git sends the trace
+                // output to that path instead of stdout (GIT_TRACE=1) or stderr (GIT_TRACE=2).
+                if (key.StartsWith("GIT_TRACE", StringComparison.OrdinalIgnoreCase))
+                {
+                    string value = processInfo.EnvironmentVariables[key];
+
+                    if (!key.Equals("GIT_TRACE")
+                        || value.IndexOfAny(Path.GetInvalidPathChars()) >= 0
+                        || !Path.IsPathRooted(processInfo.EnvironmentVariables[key]))

```

Comment:
Don't we want all the `GIT_TRACE` permutations to be able to use a fully-rooted path?  Would it make the code easier to understand by checking all the conditions for value like 1, 2, true?